code/game/machinery/landline.dm

```diff
@@ -0,0 +1,393 @@
+/obj/phone_cord
+	name = "telephone cord"
+	icon = 'icons/obj/terminals.dmi'
+	icon_state = "phone_cord"
+	anchored = TRUE
+
+/obj/landline
+	var/obj/item/telephone/linked_phone = null
+	var/obj/item/telephone/phone = null
+	var/image/phone_overlay
+	var/ringer = TRUE
+	var/ringing = FALSE
+	var/chosen_department //department we want to call
+	var/obj/landline/calling = null	//landline we are in a call with
+	var/last_call_log
+	var/obj/attached_to
+	var/phone_type = "/obj/item/telephone"
+	var/overlay_icon = 'icons/obj/terminals.dmi'
+	var/overlay_iconstate = "phone_overlay"
+	var/tether_length = 4
+	var/list/obj/phone_cord/linked_cord = list()
+	var/list/obj/item/telephone/switchboard/listening_operators = list()
+
+/obj/landline/New(var/obj/A)
```

Comment:
you've overridden the default variable which is loc, below when you create the phone you call with two args (src, src)

but do you even need 2 args? seems like you always want to attach it to the thing it's inside

Reply:
at the time i was thinking of future landline integration and how it would save repetitions of `landline.attached_to = src` right after adding a landline to other objects
let me know if i should remove `attached_to` and use `loc` in its place instead

Reply:
loc is the default so it definitely should still be there, you are free to keep the attached_to arg if you plan to have it attachable to other things (maybe default it to loc), naturally you sure ensure you use both args everywhere if you go with 2